src/cortxfs/cortxfs_internal.c

```diff
@@ -112,6 +114,14 @@ int cfs_access_check(const cfs_cred_t *cred, const struct stat *stat,
 	if (cred->uid == CFS_ROOT_UID)
 		return 0;
 
+	/* Check for matching group in group list */
+	for (i=0; i< cred->total_grps; i++) {
```

Comment:
Question: Since "total_grps" would be set by the caller (from a layer that is not cfs), should we need to check here if total_grps is less than CORTXFS_CRED_GRPS, which may lead to a memory corruption?

Reply:
In current code, where we are getting the group list from NFS client via RPC, 16 is the upper limit. So no need to check.
In case of "manage_gids" option, there can be more groups. This is not supported as of now.